Adds DomainEventHandler a priority property to control execution order

Sherpard opened the pull request:
DomainEventHandlers execution order now can be defined by using @Priority annotation.
If annotation is not present, default implementation assigns Integer.MinValue to priority

Also adds an SPI to alter the behavior of how Event handlers are managed

Sherpard committed:
SPI to intercept eventHandlers

Updated Changelog
Default Implementation uses @Priority annotation
Fixes a compilation error when built with OracleJDK 8

Bump fest-reflect from 1.4 to 1.4.1
Bumps [fest-reflect](https://github.com/alexruiz/fest-reflect) from 1.4 to 1.4.1.
- [Release notes](https://github.com/alexruiz/fest-reflect/releases)
- [Commits](https://github.com/alexruiz/fest-reflect/compare/fest-reflect-1.4...fest-reflect-1.4.1)

Bump seed.version from 3.8.2 to 3.8.5

Bumps `seed.version` from 3.8.2 to 3.8.5.

Updates `seed-core` from 3.8.2 to 3.8.5
- [Release notes](https://github.com/seedstack/seed/releases)
- [Changelog](https://github.com/seedstack/seed/blob/master/CHANGELOG.md)
- [Commits](https://github.com/seedstack/seed/compare/v3.8.2...v3.8.5)

Updates `seed-testing-junit4` from 3.8.2 to 3.8.5

adrienlauer commented:
Thank you for the nice work !

What do you think about using a`@Priority` annotation on the `DomainEventHandler`s to order them instead of an explicit method in the interface ? This annotation is the JSR standard one and is already used in various places of the framework where absolute ordering among classes is necessary.

Sherpard commented:
That was my first approach, not with standard means, but by adding a priority property @ DomainEventHandler.

Anyway, due to the nature of Events, there could be a need for different logics when handling different events, (Thinking on child events and so on), Maybe we can make a @Priority interceptor the default implementation,

Sherpard commented:
> Thank you for the nice work !
> 
> What do you think about using a`@Priority` annotation on the `DomainEventHandler`s to order them instead of an explicit method in the interface ? This annotation is the JSR standard one and is already used in various places of the framework where absolute ordering among classes is necessary.

I've updated the PR to use @Priority instead of getPriority() property from DomainEventHandler Inteface :)  way cleaner

ejemba commented on core/src/test/java/org/seedstack/business/fixtures/event/interceptors/DefaultPriorityEventHandler.java:
```diff
@@ -0,0 +1,26 @@
+package org.seedstack.business.fixtures.event.interceptors;
+
+import org.seedstack.business.domain.DomainEventHandler;
+
```
I would add javadoc explaining the role of this class.

Sherpard replied:
This is a test case, but already added some javadoc :)

ejemba commented on specs/src/main/java/org/seedstack/business/domain/DomainEventInterceptor.java:
```diff
@@ -0,0 +1,18 @@
+
+
+@SuppressWarnings("rawtypes")
+public interface DomainEventInterceptor {
```
I would add javadoc explaining the role of this interface.

Sherpard replied:
added javadoc 👍

Sherpard committed:
Added JavaDoc

adrienlauer reviewed (changes requested):
Please reflect the latest changes in the changelog.

Sherpard committed:
Merge branch 'master' into master

Sherpard committed:
Added Shed 1.1.6 as a dependency

Using PriorityUtils method at the default implementation domainEventInterceptor

Sherpard commented:
> Please reflect the latest changes in the changelog.

Already updated the changelog, :)

dependabot-preview[bot] committed:
Bump parent-internal from 3.4.6 to 3.4.7

Bumps [parent-internal](https://github.com/seedstack/poms) from 3.4.6 to 3.4.7.
- [Release notes](https://github.com/seedstack/poms/releases)
- [Changelog](https://github.com/seedstack/poms/blob/master/CHANGELOG.md)
- [Commits](https://github.com/seedstack/poms/compare/v3.4.6...v3.4.7)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Merge pull request #5 from Sherpard/dependabot/maven/org.seedstack.poms-parent-internal-3.4.7

Sherpard commented:
Already merged on #108

adrienlauer commented:
Merged through #108
